token.go

```diff
@@ -214,16 +214,61 @@ type TSTInfo struct {
 	Extensions     []pkix.Extension `asn1:"optional,tag:1"`
 }
 
+// Timestamp denots the time at which the timestamp token was created by the TSA
+//
+// Reference: RFC 3161 2.4.2
+type Timestamp struct {
+	// Value is the GenTime of TSTInfo
+	Value time.Time
+
+	// Accuracy is the Accuracy of TSTInfo
+	Accuracy time.Duration
+}
+
+// Before returns true if the lower limit of the time at which the timestamp
+// token was created is before u.
+//
+// Reference: RFC 3161 2.4.2
+func (t *Timestamp) Before(u time.Time) bool {
+	timestampLowerLimit := t.Value.Add(-t.Accuracy)
```

Comment:
Mathematically, should it be
```suggestion
	timestampUpperLimit := t.Value.Add(t.Accuracy)
```
`[2:29, 2:31]` is before `2:32` but not before `2:30`.

Reply:
Mathematically, it turns out to be a lot complicated than appears to the eye. Had a discussion with @shizhMSFT, and will update to `t.BoundedBefore(u)` and `t.BoundedAfter(u)`.